### PR TITLE
fix(ssg): RHICOMPL-2030 do not override default SSGs file, use two

### PR DESCRIPTION
**Pull request by skateman**
The `gitignore` already included the YAML file but as it was checked in to the repo, it kept updating it. By having two files, it is possible to keep one clean in the repo and the other that gets overwritten will be never checked in.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: app/services/ssg_config_downloader.rb [hunk 1]
```diff
@@ -29,6 +30,10 @@ def ssg_datastream_config_url
 
     private
 
+    def ssg_ds_file
+      File.new(File.exist?(SSG_DS_FILE_PATH) ? SSG_DS_FILE_PATH : FALLBACK_PATH)
```
Comment (akofink) [inline app/services/ssg_config_downloader.rb:34]:
```suggestion
      File.new(File.exist?(FILE_PATH) ? FILE_PATH : FALLBACK_PATH)
```

**Review by akofink — APPROVED** [review]
Thank you! This will be a great quality of life improvement for local development.

File: app/services/ssg_config_downloader.rb (hunk 1 above)
Comment (skateman) [reply app/services/ssg_config_downloader.rb]: :see_no_evil: sorry, missed this

**Comment by akofink** [comment]
You've got some failing tests. Be sure to test locally by first deleting the SSG config file.

File: app/services/ssg_config_downloader.rb [hunk 2]
```diff
@@ -3,19 +3,20 @@
 # Service for dowloading SSG configuration files
 class SsgConfigDownloader
   class << self
-    SSG_DS_FILE = File.new('config/supported_ssg.yaml')
+    FILE_PATH = 'config/supported_ssg.yaml'
```
Comment (vkrizan) [inline app/services/ssg_config_downloader.rb:6]: A bit off-topic, but can we use some temporary path? My SSD would thank you :smiley:.

File: test/services/ssg_config_downloader_test.rb [hunk 3]
```diff
@@ -2,36 +2,68 @@
 
 require 'test_helper'
 
-# A class to test the SsgConfigDownloader service
+# A class to should the SsgConfigDownloader service
```
Comment (akofink) [inline test/services/ssg_config_downloader_test.rb:5]: :thinking:

**Review by akofink — APPROVED** [review]
Thanks!

File: test/services/ssg_config_downloader_test.rb (hunk 3 above)
Comment (skateman) [reply test/services/ssg_config_downloader_test.rb]: Mass replace of the word `test` to `should` :see_no_evil:

**Pull request merged**